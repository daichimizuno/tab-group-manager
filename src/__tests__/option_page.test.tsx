import React from "react";
import renderer, { act, ReactTestRendererJSON } from "react-test-renderer";
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import Option from "../option_page";

let tree: ReactTestRendererJSON;
beforeEach(() => {
  tree = renderer.create(<Option />).toJSON() as ReactTestRendererJSON;
});

afterEach(() => {
});

describe("option_pageの試験", () => {
  describe("コンポーネントテスト", () => {
    test("スナップショットが前回の結果と一致しているかどうか", () => {
      expect(tree).toMatchSnapshot();
    });

    test("タブグループを入力できること", () => {
      act(() => {
        render(<Option />);
      });
      const input = screen.getByPlaceholderText("グループ名") as HTMLInputElement
      expect(input.value).toBe("");

      act(() => {
        userEvent.type(input,"test")
      });

      expect(input.value).toBe("test");
    });

    // test("タブの色を決められた色を選択できること", () => {});

    // test("タブグループが入力されていて作成ボタンを押すと、addNewGroup関数が呼ばれること", () => {});

    // test("タブグループが入力されていない状態で作成ボタンを押すと、何も起こらないこと", () => {});
  });
});
