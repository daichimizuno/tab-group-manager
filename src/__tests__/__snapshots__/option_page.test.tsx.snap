// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`option_pageの試験 コンポーネントテスト スナップショットが前回の結果と一致しているかどうか 1`] = `
Array [
  <div
    className="new-tab-group"
  >
    <p>
      新しいタブグループを作成
    </p>
    <input
      name="tabName"
      onChange={[Function]}
      placeholder="グループ名"
      type="text"
      value=""
    />
    <select
      onChange={[Function]}
    >
      <option
        value="grey"
      >
        grey
      </option>
      <option
        value="blue"
      >
        blue
      </option>
      <option
        value="red"
      >
        red
      </option>
      <option
        value="yellow"
      >
        yellow
      </option>
      <option
        value="green"
      >
        green
      </option>
      <option
        value="pink"
      >
        pink
      </option>
      <option
        value="purple"
      >
        purple
      </option>
      <option
        value="cyan"
      >
        cyan
      </option>
    </select>
    <button
      data-testid="createTabGroupButton"
      onClick={[Function]}
    >
      作成
    </button>
  </div>,
  <div
    className="new-url-to-group"
  >
    <p>
      URLをグループに追加
    </p>
    <input
      name="url"
      onChange={[Function]}
      placeholder="URL"
      style={
        Object {
          "width": "300px",
        }
      }
      type="text"
    />
    <select
      onChange={[Function]}
    />
    <button
      onClick={[Function]}
    >
      作成
    </button>
  </div>,
]
`;
